x/profiles/types/chain_link.go

```diff
@@ -0,0 +1,176 @@
+		return fmt.Errorf("plain text cannot be empty or blank")
+	}
+	return nil
+}
+
+func (proof Proof) Verify(unpacker codectypes.AnyUnpacker) error {
+	var pubkey cryptotypes.PubKey
+	err := unpacker.UnpackAny(proof.PubKey, &pubkey)
+	if err != nil {
+		return fmt.Errorf("failed to unpack the pubkey")
+	}
+	sig, _ := hex.DecodeString(proof.Signature)
+	if !pubkey.VerifySignature([]byte(proof.PlainText), sig) {
+		return fmt.Errorf("failed to verify the signature")
+	}
+	return nil
+}
+
+// ___________________________________________________________________________________________________________________
+
+type AddressData interface {
+	proto.Message
+	Validate() error
+	GetAddressString() string
+}
+
+func NewBech32Address(value, prefix string) *Bech32Address {
+	return &Bech32Address{Value: value, Prefix: prefix}
+}
+
+func (address Bech32Address) Validate() error {
+	if strings.TrimSpace(address.Value) == "" {
+		return fmt.Errorf("address cannot be empty or blank")
+	}
+	if strings.TrimSpace(address.Prefix) == "" {
+		return fmt.Errorf("prefix cannot be empty or blank")
+	}
+	return nil
+}
+
+func (address Bech32Address) GetAddressString() string {
+	return address.Value
+}
+
+func NewBase58Address(value, prefix string) *Base58Address {
+	return &Base58Address{Value: value}
+}
+
+func (address Base58Address) Validate() error {
+	if strings.TrimSpace(address.Value) == "" {
+		return fmt.Errorf("address cannot be empty or blank")
+	}
+	return nil
+}
+
+func (address Base58Address) GetAddressString() string {
+	return address.Value
+}
+
+func UnpackAddress(unpacker codectypes.AnyUnpacker, addressAny *codectypes.Any) (AddressData, error) {
+	var address AddressData
+	if err := unpacker.UnpackAny(addressAny, &address); err != nil {
+		return nil, err
+	}
+	return address, nil
+}
+
+// ___________________________________________________________________________________________________________________
+
+// NewChainLink is a constructor function for ChainLink
+func NewChainLink(address AddressData, proof Proof, chainConfig ChainConfig, creationTime time.Time) ChainLink {
+	addressAny, err := codectypes.NewAnyWithValue(address)
+	if err != nil {
+		panic("failed to pack address data to any type")
+	}
+	return ChainLink{
+		Address:      addressAny,
+		Proof:        proof,
+		ChainConfig:  chainConfig,
+		CreationTime: creationTime,
+	}
+}
+
+func (link ChainLink) Validate() error {
+
+	if err := link.ChainConfig.Validate(); err != nil {
+		return err
+	}
+	if err := link.Proof.Validate(); err != nil {
+		return err
+	}
+	if link.CreationTime.IsZero() {
+		return fmt.Errorf("createion time cannot be zero")
+	}
+	return nil
+}
+
+// UnpackInterfaces implements codectypes.UnpackInterfacesMessage
+func (link *ChainLink) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error {
+	if link != nil {
+		var pubKey cryptotypes.PubKey
+		if err := unpacker.UnpackAny(link.Proof.PubKey, &pubKey); err != nil {
+			return err
+		}
```

Comment:
Instead of implementing this here, we should implement the `UnpackInterfaces` method for the `Proof` object. This would make it possible to serialize/deserialize such object individually as well